### ISSUE TITLE
Bug: enum declared in nextflow script causes error

### ISSUE THREAD

**Issue by jemunro** [nextflow-io/nextflow#878]
### Expected behavior and actual behavior
Given the following script:
```
#!/usr/bin/env nextflow

enum MyEnum { FOO, BAR, BAZ }

stage = 'FOO' as MyEnum

if (stage.is(MyEnum.FOO)) {
    println('FOO Stage')
}
```
The output with nextflow version < 0.32.0 is:
```
N E X T F L O W  ~  version 0.30.0
Launching `./main.nf` [infallible_stone] - revision: f6f9591950
FOO Stage
```
However for nextflow version 0.32.0, an error occurs with the following output:
```
N E X T F L O W  ~  version 0.32.0
Launching `./main.nf` [chaotic_sammet] - revision: f6f9591950
ERROR ~ BUG! exception in phase 'conversion' in source unit '_nf_script_f1d8883a' unexpected NullpointerException

 -- Check '.nextflow.log' file for details
```
Log: [.nextflow.log](https://github.com/nextflow-io/nextflow/files/2436706/default.nextflow.log)

**Comment by pditommaso** [comment]
Thanks for reporting this. A patch will be included in the next release.

**Comment by pditommaso** [comment]
Released with version `18.10.1`.
